runtime/lua/vim/lsp/buf.lua

```diff
@@ -46,6 +46,14 @@ local function request(method, params, handler)
   return vim.lsp.buf_request(0, method, params, handler)
 end
 
+local pos_request = function(method, _params, handler)
```

Comment:
I would just manually call offset encoding instead of making this additional helper